### PR TITLE
Fix compiling when including multiple headers

### PR DESCRIPTION
**Pull request by MonsterDruide1**
For multiple reasons, the code currently does not compile when including *all* headers from the whole library. For this PR, I went through and checked the collisions or missing definitions that the compiler complained about, and fixed them. With these changes in place, I can generate a single file containing `#include`s to all headers in the whole project (`OdysseyDecomp`), so it appears like these were the only `error` collision in the entire project.

- - -
This change is [Reviewable](https://reviewable.io/reviews/open-ead/nnheaders/28)

### CODE REVIEW

**Comment by leoetlino** [comment]
Most changes look good to me except the nvn FuncPtr stuff - are you sure that's _not_ a header? cc @ThePixelGamer

**Comment by ThePixelGamer** [comment]
The loadcprocs function said it was in a header iirc, I can double check tomorrow

**Comment by ThePixelGamer** [comment]
nvnFuncImpl is supposed to be a header and it's only supposed to be included in one cpp file (maybe there should be a flag to disable building the gfx library because as far as I can tell it's the one that includes it)

**Comment by MonsterDruide1** [comment]
It's causing problems when trying to use the library in a "properly linked program", as the header might be included multiple times. As VSCode already complains about it:
```
Function 'nvnLoadCProcs' defined in a header file; function definitions in header files can lead to ODR violations (fix available)
```

**Comment by ThePixelGamer** [comment]
Would marking the function as inline avoid that?

**Comment by MonsterDruide1** [comment]
For the function, yes. However, all "variables" above also have a similar error message attached:
```
Variable 'pfnc_nvnCommandBufferSignalEvent' defined in a header file; variable definitions in header files can lead to ODR violations
```
Is there a similar thing like `inline` for the function? Also, for doing `inline` on that function, wouldn't that mean that the function generates no symbol at all?

**Comment by ThePixelGamer** [comment]
There is but it's C++17 and I'm not sure about the symbol? Maybe a cpp file is the best route here but maybe another potential option is just disabling that error?

**Comment by leoetlino** [comment]
Are we sure that the variables weren't declared (extern) in the header but defined in a .cpp?

**Comment by fruityloops1** [comment]
> Would marking the function as inline avoid that?

This function is not inlined in the first place, it should be defined in it's own TU.

**Comment by ThePixelGamer** [comment]
They're definitely defined in the header but I'm not entirely sure how they did it, same thing with the cpp version (which I'm surprised isn't giving any errors for you?). Honestly feel free to change this if it's needed but I believe this is closer to what was originally written.

**Comment by MonsterDruide1** [comment]
I just took the "easiest path" here to fix the error, doing it differently might be a lot better.

@leoetlino They are already declared with `extern` in `nvn_FuncPtrBase.h`.

**Comment by leoetlino** [comment]
I don't understand how nvn_FuncPtrImpl.h is supposed to be used. If it's supposed to be included in a single translation unit, and we've seen it included in a game-specific TU (in Diablo), then how does the include in gfx_NvnHelper.cpp even work?

**Comment by ThePixelGamer** [comment]
Tbh the design doesn't make sense to me either, I just tried my best to replicate what I believe is their setup. If it's better for usability we can just stick it in its own TU (like how this PR does it) and use a cmake flag on whether or not to include it.

**Comment by MonsterDruide1** [comment]
So, what should be adjusted in this PR to get this merged?

**Comment by ThePixelGamer** [comment]
Copying here for the sake of prosperity : either only include those impl headers once (and maybe we could supply a flag in the cmake to enforce that) or move them to a TU (like how this PR does it) and choose which ones to use with a cmake flag

File: src/NintendoSDK/gfx/detail/gfx_NvnHelper.cpp [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 #include <nn/gfx/gfx_SwapChainInfo.h>
 #include <nn/gfx/gfx_TextureInfo.h>
-#include <nvn/nvn_FuncPtrImpl.h>
+#include <nvn/nvn_FuncPtrBase.h>
```
Comment (MonsterDruide1) [inline src/NintendoSDK/gfx/detail/gfx_NvnHelper.cpp:5]: Which one is correct here, should it actually include the `Impl`? Or should I change it back, and you will take care of it in the next PR?

Comment (ThePixelGamer) [reply src/NintendoSDK/gfx/detail/gfx_NvnHelper.cpp:5]: It should be fine, it would've been changed in the next PR anyways

**Pull request merged**